crates/bevy_animation_rig/src/skinned_mesh.rs

```diff
@@ -0,0 +1,369 @@
+                .iter()
+                .map(|&entity| SkinnedMeshJoint {
+                    entity,
+                    transform: Mat4::IDENTITY,
+                })
+                .collect(),
+        }
+    }
+
+    pub fn update_joint_transforms(
+        &mut self,
+        inverse_bindposes_assets: &Res<Assets<SkinnedMeshInverseBindposes>>,
+        global_transform_query: &Query<&GlobalTransform>,
+    ) {
+        let inverse_bindposes = inverse_bindposes_assets
+            .get(self.inverse_bindposes.clone())
+            .unwrap();
+
+        for (joint, &inverse_bindpose) in self.joints.iter_mut().zip(inverse_bindposes.0.iter()) {
+            let global_transform = global_transform_query.get(joint.entity).unwrap();
+            joint.transform = global_transform.compute_matrix() * inverse_bindpose;
+        }
+    }
+}
+
+impl MapEntities for SkinnedMesh {
+    fn map_entities(&mut self, entity_map: &EntityMap) -> Result<(), MapEntitiesError> {
+        for joint in &mut self.joints {
+            joint.entity = entity_map.get(joint.entity)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl RenderResource for SkinnedMesh {
+    fn resource_type(&self) -> Option<RenderResourceType> {
+        Some(RenderResourceType::Buffer)
+    }
+
+    fn write_buffer_bytes(&self, buffer: &mut [u8]) {
+        let transform_size = std::mem::size_of::<[f32; 16]>();
+
+        for (index, joint) in self.joints.iter().enumerate() {
+            joint.transform.write_buffer_bytes(
+                &mut buffer[index * transform_size..(index + 1) * transform_size],
+            );
+        }
+    }
+
+    fn buffer_byte_len(&self) -> Option<usize> {
+        Some(self.joints.len() * std::mem::size_of::<[f32; 16]>())
+    }
+
+    fn texture(&self) -> Option<&Handle<Texture>> {
+        None
+    }
+}
+
+impl RenderResources for SkinnedMesh {
+    fn render_resources_len(&self) -> usize {
+        1
+    }
+
+    fn get_render_resource(&self, index: usize) -> Option<&dyn RenderResource> {
+        (index == 0).then(|| self as &dyn RenderResource)
+    }
+
+    fn get_render_resource_name(&self, index: usize) -> Option<&str> {
+        (index == 0).then(|| buffer::JOINT_TRANSFORMS)
+    }
+
+    // Used to tell GLSL to use storage buffer instead of uniform buffer
+    fn get_render_resource_hints(&self, index: usize) -> Option<RenderResourceHints> {
+        (index == 0).then(|| RenderResourceHints::BUFFER)
+    }
+
+    fn iter(&self) -> RenderResourceIterator {
+        RenderResourceIterator::new(self)
+    }
+}
+
+/// Store data for each joint belongs to the [`SkinnedMesh`]
+#[derive(Debug, Clone)]
+pub struct SkinnedMeshJoint {
+    pub entity: Entity,
+    pub transform: Mat4,
+}
+
+/// Manually implement [`bevy_reflect::Reflect`] for [`SkinnedMeshJoint`] to work around an issue,
+/// where spawning a scene with a component containings a vector of structs would result in runtime panic.
+unsafe impl Reflect for SkinnedMeshJoint {
```

Comment:
I'm curious, is deriving `Reflect` on the `SkinnedMeshJoint` enough? 
(Instead of manually implementing it?)

Reply:
It's not just normal manual implementation but more like a monkey-patch. I patched the `clone_value()` method.